### PR TITLE
fix renaming

### PR DESCRIPTION
**Pull request by Nik-Sch**

### CODE REVIEW

**Comment by g0t00** [comment]
test cases would be so nice. But I am working on it ;)

File: lib/languageFeatures/findReferencesHandler.ts [hunk 1]
```diff
@@ -56,16 +56,18 @@ export async function prepareRenameHandler(params: TextDocumentPositionParams) {
...
 export async function renameHandler(params: RenameParams) {
   const references = (await findReferences(params)).map(reference => {
-    if (implementsIReferencable(reference) && implementsIHasLexerToken(reference)) {
+    if (implementsIHasLexerToken(reference)) {
       return reference.lexerToken;
     }
     return reference;
-  });
-  // console.log(references.map(reference => `${reference.range.start.line} ${reference.range.start.character}`));
+  }).filter((r, i, s) => s.indexOf(r) === i);
```
Comment (g0t00) [inline lib/languageFeatures/findReferencesHandler.ts:70]: What is this? Explain maybe

Comment (g0t00) [reply lib/languageFeatures/findReferencesHandler.ts:70]: Ah, the parameters are default filter, then it is always true oO?
Ah it picks only the first If the exact same is in it twice? why not use Set?

**Review by g0t00 — CHANGES_REQUESTED** [review]

File: lib/languageFeatures/findReferencesHandler.ts (hunk 1 above)
Comment (Nik-Sch) [reply lib/languageFeatures/findReferencesHandler.ts]: I made it clearer

**Review by g0t00 — APPROVED** [review]

**Pull request merged**